platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileMetadata.java

```diff
@@ -84,13 +86,16 @@ public DataFileMetadata(
             final long dataItemCount,
             final int index,
             final Instant creationDate,
-            final long serializationVersion) {
+            final long serializationVersion,
+            final int compactionLevel) {
         this.fileFormatVersion = fileFormatVersion;
         this.dataItemValueSize = dataItemValueSize;
         this.dataItemCount = dataItemCount;
         this.index = index;
         this.creationDate = creationDate;
         this.serializationVersion = serializationVersion;
+        assert compactionLevel >= 0 && compactionLevel < 127;
```

Comment:
Hard-coded value 127 would be better to turn into a documented constant like `MAX_COMPACTION_LEVEL`

Reply:
fixed